File: backend/nestjs/clean-architecture/minimal/src/modules/auth/application/dto/login-response.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { UserDto } from '@/modules/user-management/application/dto/user.dto';

export class LoginResponseDto {
  @ApiProperty({ description: 'JWT access token' })
  token: string;

  @ApiProperty({ description: 'Refresh token' })
  refreshToken: string;

  @ApiProperty({ description: 'Token type', default: 'Bearer' })
  tokenType: string;

  @ApiProperty({ description: 'Access token expiration time in seconds' })
  expiresIn: number;

  @ApiProperty({ description: 'Refresh token expiration time' })
  refreshTokenExpiresAt: Date;

  @ApiProperty({ description: 'User information' })
  user: UserDto;

  constructor(
    token: string,
    refreshToken: string,
    expiresIn: number,
    refreshTokenExpiresAt: Date,
    user: UserDto,
    tokenType: string = 'Bearer',
  ) {
    this.token = token;
    this.refreshToken = refreshToken;
    this.tokenType = tokenType;
    this.expiresIn = expiresIn;
    this.refreshTokenExpiresAt = refreshTokenExpiresAt;
    this.user = user;
  }
}

File: backend/dotnet/clean-architecture/full/src/Core/App.Template.Application/DTOs/Auth/LoginResponseDto.cs
namespace AppTemplate.Application.DTOs.Auth;

/// <summary>
/// Login response with JWT token
/// </summary>
public record LoginResponseDto
{
    /// <summary>
    /// JWT access token
    /// </summary>
    public string AccessToken { get; init; } = string.Empty;

    /// <summary>
    /// Token expiration time in seconds
    /// </summary>
    public int ExpiresIn { get; init; }

    /// <summary>
    /// Refresh token for obtaining new access tokens
    /// </summary>
    public string? RefreshToken { get; init; }

    /// <summary>
    /// User information
    /// </summary>
    public UserInfoDto? User { get; init; }
}

/// <summary>
/// Refresh token response (no user info)
/// </summary>
public record RefreshResponse
{
    /// <summary>
    /// New JWT access token
    /// </summary>
    public string AccessToken { get; init; } = string.Empty;

    /// <summary>
    /// Token expiration time in seconds
    /// </summary>
    public int ExpiresIn { get; init; }

    /// <summary>
    /// New refresh token
    /// </summary>
    public string RefreshToken { get; init; } = string.Empty;
}
